src/python/pants/jvm/subsystems.py

```diff
@@ -98,3 +98,22 @@ class JvmSubsystem(Subsystem):
         ),
         advanced=True,
     )
+    # See https://github.com/pantsbuild/pants/issues/14937 for discussion of one way to improve
+    # our behavior around cancellation with nailgun.
+    nailgun_remote_cache_speculation_delay = IntOption(
+        default=1000,
```

Comment:
@somdoron : As a nod to this setting actually needing to exist, I've set a default here. It's arbitrary though: if you have time to experiment with a few values and see what works best for you, we'd love a patch to change it further.

Reply:
I used the same number in our repository, so looks good to me.